spec/units/token_parser_spec.rb

```diff
@@ -0,0 +1,130 @@
+require 'spec_helper'
+require 'gitsh/token_parser'
+
+describe Gitsh::TokenParser do
+  describe '.parse' do
+    it 'parses Git commands with no arguments' do
+      command = double(:command)
+      allow(Gitsh::Commands::Factory).to receive(:build).and_return(command)
+
+      result = described_class.parse(tokens([:WORD, 'commit'], [:EOS]))
+
+      expect(result).to eq command
+      expect(Gitsh::Commands::Factory).to have_received(:build).with(
+        Gitsh::Commands::GitCommand,
+        hash_including(command: 'commit', args: []),
+      )
+    end
+
+    it 'parses Git commands broken into multiple words' do
```

Comment:
The intention behind this was to support that most general-purpose shells support, where you can mash a quoted word and an unquoted word together, and they're interpreted as a single word. For example, in bash, sh, and zsh, you can run `"l"s` or `ls` or `l's'` and they're all parsed the same way.